File: core/channels/channel_info.cpp
#include "channels/channel_info.h"

namespace Neuro {

const ChannelInfo ChannelInfo::Signal = ChannelInfo(Type::Signal);
const ChannelInfo ChannelInfo::Battery = ChannelInfo(Type::Battery);
const ChannelInfo ChannelInfo::ElectrodesState = ChannelInfo(Type::ElectrodesState);
const ChannelInfo ChannelInfo::Respiration = ChannelInfo(Type::Respiration);
const ChannelInfo ChannelInfo::MEMS = ChannelInfo(Type::MEMS);
const ChannelInfo ChannelInfo::Angle = ChannelInfo(Type::Angle);
const ChannelInfo ChannelInfo::ConnectionStats = ChannelInfo(Type::ConnectionStats);
const ChannelInfo ChannelInfo::Resistance = ChannelInfo(Type::Resistance);
const ChannelInfo ChannelInfo::Pedometer = ChannelInfo(Type::Pedometer);

ChannelInfo::ChannelInfo(ChannelInfo::Type type):
    mType(type),
    mName(typeToName(mType)){
}

ChannelInfo::ChannelInfo(Type type, std::string &&name) noexcept :
    mType(type),
    mName(std::move(name)){
}

ChannelInfo::ChannelInfo(Type type, const std::string &name):
    mType(type),
    mName(name){
}

std::string ChannelInfo::getName() const {
    return mName;
}

void ChannelInfo::setName(std::string &&name) noexcept {
    mName = std::move(name);
}

void ChannelInfo::setName(const std::string &name) {
    mName = name;
}

ChannelInfo::Type ChannelInfo::getType() const noexcept {
    return mType;
}

bool ChannelInfo::operator==(const ChannelInfo &rhs) const noexcept {
    return (mType == rhs.mType) && (mName == rhs.mName);
}

bool ChannelInfo::operator!=(const ChannelInfo &rhs) const noexcept {
    return !(*this == rhs);
}

}

File: wrappers/android/android-neurosdk/src/main/java/com/neuromd/neurosdk/channels/ChannelInfo.java
package com.neuromd.neurosdk.channels;

public class ChannelInfo {
    private String mName;
    private final ChannelType mType;
    private final long mIndex;

    public ChannelInfo(ChannelType type){
        mType = type;
        mName = type.name();
        mIndex = 0;
    }

    public ChannelInfo(ChannelType type, String name){
        mType = type;
        mName = name;
        mIndex = 0;
    }

    public ChannelInfo(ChannelType type, String name, long index){
        mType = type;
        mName = name;
        mIndex = index;
    }

    public ChannelInfo(ChannelInfo rhs){
        mName = rhs.getName();
        mType = rhs.getType();
        mIndex = rhs.getIndex();
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public ChannelType getType() {
        return mType;
    }

    public long getIndex()    {
        return mIndex;
    }
}
